### PR TITLE
Feat(Support attachments): pass in attachments

### LINKED ISSUE #192: [FEATURE] Support Attachments
**Is your feature request related to a problem? Please describe.**
I'd like to support media attachments in AgentMark.

**Describe the solution you'd like**
Add a tag or set of tags that support media attachments. Note, see the "ExtractText" plugin for a reference.

**Additional context**
Note: https://ai-sdk.dev/docs/reference/ai-sdk-core/generate-text#messages.core-user-message.content see how vercel ai supports attachments

### PR DESCRIPTION
**Pull request by Jawad-A02**
## Description

Support passing in attachments to text and object generation.

1. Changed the messages type to support the defined user messages on [vercel AI SDK:](https://ai-sdk.dev/docs/reference/ai-sdk-core/generate-object#messages.core-user-message.content)
2. Changed the handling for extracting user messages in templateDX 

Fixes #192

## How Has This Been Tested?

So far has done the testing primarily through the index file but I am working on the test files right now
Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/agentmark/src/template_engines/templatedx.ts [hunk 1]
```diff
@@ -57,9 +111,86 @@ export class ExtractTextPlugin extends TagPlugin {
...
+          break;
+      }
+    });
+
+    return Promise.resolve({ name: tagName, content: extractedParts });
+  }
 }
 
-TagPluginRegistry.register(new ExtractTextPlugin(), ["User", "System", "Assistant"]);
+TagPluginRegistry.register(new ExtractTextPlugin(), [
```
Comment (kocobee) [inline packages/agentmark/src/template_engines/templatedx.ts:189]: Chatted offline. We'll want separate a separate plugin for `<File>/<Image>`

File: packages/agentmark/src/template_engines/templatedx.ts [hunk 2]
```diff
@@ -29,25 +38,41 @@ export class ExtractTextPlugin extends TagPlugin {
...
           // @ts-ignore
           children: flattenedChildren,
         });
-        resolve({ content: extractedText.trim(), name: tagName });
+
+        const mediaParts = scope.getShared("__puzzlet-mediaParts") || [];
+        const hasMediaContent = tagName === "User" && mediaParts.length;
+        const content = hasMediaContent
+          ? [{ type: "text", text: extractedText.trim() }, ...media]
+          : extractedText.trim();
```
Comment (Jawad-A02) [inline packages/agentmark/src/template_engines/templatedx.ts:64]: So this will mean all the text will be combined together as one object of type `text`, but it doesn't care about the order or separating text. Now we could fix it by adding more tags to the extractMediaPlugin but wanted to ask if I should do it first

File: packages/agentmark/test/fixtures/attachments.prompt.mdx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    type: object
+    properties:
+      answer:
+        type: string
+    required: ["answer"]
+---
+
+<System>
+You are a helpful math tutor.
+</System>
```
Comment (Jawad-A02) [inline packages/agentmark/test/fixtures/attachments.prompt.mdx:16]: I manually tested adding a `ImageAttachment` and `ImageAttachment` and it did throw an error. Now For test suites, should I add another mdx file just for testing negative scenarios?

File: packages/agentmark/src/template_engines/templatedx.ts (hunk 2 above)
Comment (kocobee) [reply packages/agentmark/src/template_engines/templatedx.ts:64]: Hmm, if it's not too much effort, I agree we create the text/file/image array in the order that it is within the User tag. So multiple text tags ordered properly probably makes the most sense.

Comment (Jawad-A02) [reply packages/agentmark/src/template_engines/templatedx.ts:66]: How would that work? cause the extractPlugin only works on the custom tags and not native tags like `paragraph` and `text` which are normal text and `inputProps`?

File: packages/agentmark/src/template_engines/templatedx.ts [hunk 4]
```diff
@@ -59,7 +84,55 @@ export class ExtractTextPlugin extends TagPlugin {
...
+      if (image) {
+        mediaParts.push({
+          type: "image",
+          image,
+          ...(mimeType && { mimeType }),
+        });
+      }
+    } else if (tagName === "FileAttachment") {
+      const { data, mimeType } = props;
+      if (data && mimeType) {
```
[the remark this answers is not in the document]
Comment (Jawad-A02) [reply packages/agentmark/src/template_engines/templatedx.ts:132]: Never mind, I figured it out. I needed to ensure that the` image prop` is required for `ImageAttachment,` and both `data and mimeType` are required for `FileAttachment.` This works correctly when the props are actually missing like `<ImageAttachment /> instead of <ImageAttachment image="link" />`

However, I was confused because passing something like image={props.imageLink} used to fail, and I thought the error shouldn't be triggered. But I realized that during formatting, the values get populated — so they were just undefined at parse time, not truly missing.

So I needed two separate checks:
1. One to ensure the required props are defined (even if empty or placeholders).
1. Another to ensure that media parts are only added once the values are actually populated during formatting.

File: packages/agentmark/src/template_engines/templatedx.ts (hunk 2 above)
Comment (kocobee) [reply packages/agentmark/src/template_engines/templatedx.ts:66]: Let's keep as one text object for now. We can make a new feature/fix for multiple text messages later

Comment (kocobee) [reply packages/agentmark/src/template_engines/templatedx.ts:66]: Hmm. Is it possible to dump all of the contents into a text object, until we hit a file/image, and then add the new file/image to the next index of the array? If this is complex, or there's not a simple way, we can address this in a follow-up story later. I think it's lower priority.
